conan/tools/meson/meson.py

```diff
@@ -1,40 +1,36 @@
 import os
 
 from conan.tools.build import build_jobs
-from conan.tools.cross_building import cross_building
 from conan.tools.meson import MesonToolchain
 
 
 class Meson(object):
-    def __init__(self, conanfile, build_folder='build'):
+    def __init__(self, conanfile):
         self._conanfile = conanfile
-        self._build_folder = build_folder
 
-    @property
-    def _build_dir(self):
-        build = self._conanfile.build_folder
-        if self._build_folder:
-            build = os.path.join(self._conanfile.build_folder, self._build_folder)
-        return build
-
-    def configure(self, source_folder=None):
-        source = self._conanfile.source_folder
-        if source_folder:
-            source = os.path.join(self._conanfile.source_folder, source_folder)
+    def configure(self,  build_script_folder=None):
+        meson_source_folder = self._conanfile.source_folder
+        meson_build_folder = self._conanfile.build_folder
+        if build_script_folder:
+            meson_source_folder = os.path.join(self._conanfile.source_folder, build_script_folder)
 
         cmd = "meson setup"
-        if cross_building(self._conanfile):
-            cmd += ' --cross-file "{}"'.format(MesonToolchain.cross_filename)
+        generators_folder = self._conanfile.generators_folder
+        cross = os.path.join(generators_folder, MesonToolchain.cross_filename)
```

Comment:
I think it is reliable enough. of course, some process could create or delete the file in the middle of conan's execution, but realistically it's a low chance, and anyway, further execution wouldn't be correct.

Reply:
The problem with using here a hardcoded ``if cross_building()`` is that if the ``MesonToolchain`` decides (the user can do it) that it is not cross-building, this will be wrong. The build helpers like ``Meson`` should not decide things just execute what the toolchain has decided to do.

Reply:
could it be problematic in some flows, e.g. if I run several toolchain executions for multiple profiles?